app/src/main/java/com/group80/uoftinder/firebase/realtime/UserRealtimeDbFacade.java

```diff
@@ -0,0 +1,77 @@
+package com.group80.uoftinder.firebase.realtime;
+
+import com.group80.uoftinder.entities.User;
+
+import java.util.List;
+
+/**
+ * A Facade for methods regrading User in realtime database. All methods are constants, and no
+ * instances of this Facade should be created.
+ */
+public class UserRealtimeDbFacade {
+    private static final String DEFAULT_USER_TYPE = "academic";
+
+    /**
+     * Upload a user to the realtime database, based on the user type and user id. The user is
+     * stored under "users" / user.userType / user.uid
+     *
+     * @param user the user to be stored in the database
+     */
+    public static void uploadUser(User user) {
+        ucClassWriter<User> userWriter = new ucClassWriter<>();
```

Comment:
This is a problem about scope. Instance variables of an Object (also called fields) and vairables created inside methods have different sope and life cycles -- private fields are SHARED WITH IN THE CLASS, is created along with the class, and dies with the class. This means that 
1. Any changes made to a class field will be carried on to the future. In some occations we may need this, but definitely not here -- I definitely do not want two saparate upload tasks affect eachother (since I'm mutating the writer). 
2. Since all my methods are static, any fields called inside them must also be static. This means that my fields will be there throught the life cycle of the entire app -- it will not be collected by the java garbase collector to free up resources. This means that it's taking up unnecessary resources, a.k.a. memory leak. 

Now, back to what I'm doing, those two writers are declared and created INSIDE a meththod, as non-static variable. Their scope is the method itself (or like the brackets `{}` that surrounds the method body), meaning they are completely independent from each other -- I can change stuff however I want in one method, and everything will still be fresh and clean when I instanciate another instance inside the other method. Also their life cycles are significantly shorter -- when the method is complete, java junk collector will know that these resources shall no longer be used, thus freeing them to prevent memory leak.

Reply:
Okay, sounds good! Just making sure that this implementation was justified.